src/drivers/intel/ssp/ssp.c

```diff
@@ -125,6 +125,111 @@ static int ssp_context_restore(struct dai *dai)
 	return 0;
 }
 
+static int ssp_mclk_prepare_enable(struct dai *dai)
+{
+	struct ssp_pdata *ssp = dai_get_drvdata(dai);
+	struct sof_ipc_dai_config *config = &ssp->config;
+	int ret;
+
+	if (ssp->clk_active & SSP_CLK_MCLK_ACTIVE)
```

Comment:
one note that the MCLK actually doesn't belong to any of the SSP port, there could be multiple MCLKs, the mapping relation between SSP and MCLK is M:N, so attaching the _MCLK_ACTIVE to a specific SSP port looks problematic to me.

Reply:
After think it more, this should fine, SSP_CLK_MCLK_ACTIVE means if the MCLK required for this SSP codec satisfied or not. But please note that even (ssp->clk_active & SSP_CLK_MCLK_ACTIVE) is false, the MCLK pin could still be active as other SSP port could requires the same mclk at the same time.

Reply:
this PR does not change anything wrt to the multiple use of the MCLK by different SSPs.